controllers/user.controller.js

```diff
@@ -0,0 +1,108 @@
+const {ObjectId} = require("mongoose").Types;
+
+class UserController {
+
+        /**
+     * 
+     * ### Description
+     * Creates a user profile with the data that the user passes in the body.
+     * @param {Request} req 
+     * @param {Response} res 
+     * @param {Next} next 
+     */
+    static createUserProfile = async(req, res, next)=>{
+        try{
+            let data = req.body;
+            if(Object.keys(data).length == 0) throw new Error("No data passed to create user profile");
+            let user = User.create(data);
+            JSONResponse.success(res, "User profile successfully created", user, 201);
+        }catch(error){
+            JSONResponse.error(res, "Error createing user profile", error, 400);
+        }
+    }
+
+        /**
+     * 
+     * ### Description
+     * Gets the user profile for a single user with the id that is passed in as a parameter, then updates the user with the data that is passed in the body.
+     * @param {Request} req 
+     * @param {Response} res 
+     * @param {Next} next 
+     */
+    static updateUserProfile = async(req, res, next)=>{
+        try{
+            let data = req.body;
+            let id = req.params.id;
+            if(!ObjectId.isValid(id)) throw new Error("Invalid ID was passed as a parameter");
+            if(Object.keys(data).length == 0) {
+                return JSONResponse.success(res, "No data passed, file not updated",{}, 200);
+            }
+            let user = User.findByIdAndUpdate(id); 
+            JSONResponse.success(res, "User updated successfully", user, 200);
+        }catch(error){
+            JSONResponse.error(res, "Unable to update user profile", error, 400);
+        }
+    }
+
+
+        /**
+     * 
+     * ### Description
+     * Gets all users that have created a profile in the database.
+     * @param {Request} req 
+     * @param {Response} res 
+     * @param {Next} next 
+     */
+    static getAllUsers = async (req, res, next)=>{
+        try{
+            let users = User.find();
+            JSONResponse.success(res, "Retrieved all users", users, 200);
+        }catch(error){
+            JSONResponse.error(res, "Unable to find users", error, 500);
+        }
+    }
+
+    /**
+     * 
+     * ### Description
+     * Gets the user profile for a single user with the id that is passed in as a parameter and deletes it, returning the user that was deleted.
+     * @param {Request} req 
+     * @param {Response} res 
+     * @param {Next} next 
+     */
+    static deleteUserProfile = async(req, res, next)=>{
+        try{
+            let id = req.params.id;
+            if(!ObjectId.isValid(id)) throw new Error("ID does not match any user profile in database");
+            let user = User.findByIdAndDelete(id);
+            JSONResponse.success(res, "Successfully deleted user", user, 203);
+        }catch(error){
+            JSONResponse.error(res, "Unable to delete user",error, 500 );
```

Comment:
avoid 500 error codes. this indicates that the server has crashed. 
if the user is not found use 404.